src/main/java/uk/gov/register/service/ItemValidator.java

```diff
@@ -64,6 +66,16 @@ private void validateFieldsValue(JsonNode inputEntry, Map<String, Field> fields)
         });
     }
 
+    private void dataTypeBasedValidation(final JsonNode inputEntry, final String fieldName, final Datatype datatype,
```

Comment:
I feel that this method should exist in the implementation of the existing `DaterimeDataType.isValid()` method (there is a TODO to implement validation here). That way validation would be consistent for all datatypes as each has an `isValid()` method implemented.